mindsdb/utilities/config.py

```diff
@@ -16,13 +18,58 @@ def __init__(self, config_path):
         else:
             raise TypeError('Argument must be string representing a file path <Later on to be switched to file path and/or database connection info>')
 
+    def _migrate(self):
+        def m1_0(config):
+            if 'default_clickhouse' in config['integrations'] and 'type' not in config['integrations']['default_clickhouse']:
+                config['integrations']['default_clickhouse']['type'] = 'clickhouse'
+            if 'default_mariadb' in config['integrations'] and 'type' not in config['integrations']['default_mariadb']:
+                config['integrations']['default_mariadb']['type'] = 'mariadb'
+            if 'datasources' in config['api']['mysql']:
+                del config['api']['mysql']['datasources']
+            config['config_version'] = '1.1'
+            return config
+
+        migrations = {
+            '1.0': m1_0
+        }
+
+        current_version = self._parse_version(self._config['config_version'])
```

Comment:
there is packaging,version.Version class that takes a version string in constructor and supports `<=, ==, >=`operators, it might be useful here

Reply:
yes, 'packaging' good for version comparison. But adding additional dependency for one function can be a bit overhead. So i made my own :)